client/mapCatalog.js

```diff
@@ -74,6 +69,14 @@ var MapCatalog = function (maps) {
     }
   }
 
+  this.prepareNextWorld = function(world){
+    currentWorldIdx++;
+    if(currentWorldIdx >= worlds.length){
+      currentWorldIdx = 0;
+    }
+    return this.prepareCurrentWorld(world);
```

Comment:
this doesn't need to use a callback because nothing is happening in this function. maybe this should actually prepare the next world by reading it from the source and storing that in a 'nextWorld' variable somewhere, then when it's time to switch it can just start using that instead of having to do a bunch of processing right when the round is supposed to start ?

Reply:
if you do make this function load the world from the file, then keep the callback. but it will make the callback on the other function unnecessary - the file reading is the thing that necessitates a callback because it starts a new thread or process or something.